modRlp/src/org/aion/rlp/Utils.java

```diff
@@ -66,10 +79,14 @@
     }
 
     static byte[] hexEncode(byte[] in) {
+        return hexEncode(in, false);
+    }
+
+    static byte[] hexEncode(byte[] in, boolean withTerminatorByte) {
         if (in == null) {
             return null;
         }
-        byte[] ret = new byte[in.length * 2];
+        byte[] ret = new byte[in.length * 2 + (withTerminatorByte ? 1 : 0)];
```

Comment:
Why do we add that extra byte at the end that never gets populated?  (Don't think the loop below ever writes into ret[in.length*2+1])

Reply:
I did this as an optimization to avoid creating and copying several byte arrays. This way the proper length array gets created from the start. Since this is just a utility, I prefer keeping it as is.